dockermodelrunner/desktop/desktop.go

```diff
@@ -0,0 +1,489 @@
+		if expanded, err := c.fullModelID(model); err == nil {
+			model = expanded
+		}
+	}
+
+	reqBody := OpenAIChatRequest{
+		Model: model,
+		Messages: []OpenAIChatMessage{
+			{
+				Role:    "user",
+				Content: prompt,
+			},
+		},
+		Stream: true,
+	}
+
+	jsonData, err := json.Marshal(reqBody)
+	if err != nil {
+		return fmt.Errorf("marshal request: %w", err)
+	}
+
+	chatCompletionsPath := inference.InferencePrefix + "/v1/chat/completions"
+	resp, err := c.doRequest(
+		http.MethodPost,
+		chatCompletionsPath,
+		bytes.NewReader(jsonData),
+	)
+	if err != nil {
+		return c.handleQueryError(err, chatCompletionsPath)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("chat with %s status=%d body=%s", model, resp.StatusCode, body)
+	}
+
+	scanner := bufio.NewScanner(resp.Body)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			continue
+		}
+
+		if !strings.HasPrefix(line, "data: ") {
+			continue
+		}
+
+		data := strings.TrimPrefix(line, "data: ")
+
+		if data == "[DONE]" {
+			break
+		}
+
+		var streamResp OpenAIChatResponse
+		if err := json.Unmarshal([]byte(data), &streamResp); err != nil {
+			return fmt.Errorf("unmarshal stream response: %w", err)
+		}
+
+		if len(streamResp.Choices) > 0 && streamResp.Choices[0].Delta.Content != "" {
+			chunk := streamResp.Choices[0].Delta.Content
+			fmt.Print(chunk)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("read response stream: %w", err)
+	}
+
+	return nil
+}
+
+// Remove removes a model from the Docker Model Runner.
+func (c *Client) Remove(models []string, force bool) (string, error) {
+	modelRemoved := ""
+	for _, model := range models {
+		// Check if not a model ID passed as parameter.
+		if !strings.Contains(model, "/") {
+			if expanded, err := c.fullModelID(model); err == nil {
+				model = expanded
+			}
+		}
+
+		// Construct the URL with query parameters
+		removePath := fmt.Sprintf("%s/%s?force=%s",
+			inference.ModelsPrefix,
+			model,
+			strconv.FormatBool(force),
+		)
+
+		resp, err := c.doRequest(http.MethodDelete, removePath, nil)
+		if err != nil {
+			return modelRemoved, c.handleQueryError(err, removePath)
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			if resp.StatusCode == http.StatusNotFound {
+				return modelRemoved, fmt.Errorf("no such model: %s", model)
+			}
+			var bodyStr string
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				bodyStr = fmt.Sprintf("(failed to read response body: %v)", err)
+			} else {
+				bodyStr = string(body)
+			}
+			return modelRemoved, fmt.Errorf("removing %s failed with status %s: %s", model, resp.Status, bodyStr)
+		}
+		modelRemoved += fmt.Sprintf("Model %s removed successfully\n", model)
+	}
+	return modelRemoved, nil
+}
+
+// URL returns the URL for the Docker Model Runner.
+func URL(path string) string {
```

Comment:
question: Is this out date as it doesn't work for me, changing it to be `http://model-runner.docker.internal/` without the experimental prefix fixes that.